docs/design/provisioning.md

```diff
@@ -0,0 +1,223 @@
+
+`ConfigMap` objects live in a namespace, so the administrator will create a namespace specifically to hold provisioner configuration.  This "system namespace" may contain many types of ConfigMaps, not all of which are provisioners, so a specific label must be applied to provisioners to identify them.
+
+The  namespace and label are required by kube-controller-manager so that the provisioner controller can query the correct namespace and find the provisioner configs.
+
+Admins will pass the namespace and label to the controller via CLI flag:
+
+```
+CLI flags:
+
+--pv-provisioner-namespace=foobar
+--pv-provisioner-label="type=provisioner-config
+
+example usage:
+
+kubectl --namespace=system-namespace get configmap -l type=provisioner-config
+
+```
+
+
+### Provisioner Configuration
+
+Use case:  Configure provisioners for GCE Persistent Disks in two zones (east and west) and two volume types (standard and solid state).  The admin chooses to use the term "storage-class" as a stand-in for volume type (speed) and uses the labels "gold" and "silver" for fast and slow.
+
+A distinct set of labels on a provisioner is the "storage class".  The use case requires 4 provisioners: gold+east, gold+west, silver+east, silver+west.   Each provisioner's "parameters" map would contain the values for volume type and zone.  The plugin uses the parameters to create that one type of PersistentVolume.
+
+One configuration is shown below.  The remainder can be found [here](provisioning-examples.json).
+
+```
+  {
+    "kind": "ConfigMap",
+    "apiVersion": "v1",
+    "metadata": {
+      "name": "gold-east",
+      "namespace": "kubernetes.io/provisioners",
+      "labels": {
+        "type": "provisioner-config",
+        "storage-class": "gold",
+        "zone": "us-east"
+      }
+    },
+    "data": {
+      "plugin-name": "kubernetes.io/gce-pd",  
+      "parameters": "{'volumeType':'ssd', 'zone':'us-east'}"  // OPAQUE JSON MAP OF PARAMS AS DEFINED BY THE PLUGIN
+    }
+  }
+  
+```
+
+End users request specific storage by using a PersistentVolumeSelector on their claim.  This example shows a request for "gold" storage in the east zone.
+
+```
+{
+  "kind": "PersistentVolumeClaim",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "claim2",
+    "namespace": "end-user-namespace"
+  },
+  "spec": {
+    "accessModes": [
+      "ReadWriteOnce"
+    ],
+    "persistentVolumeSelector":{
+    	"matchLabels":{
+    		"storage-class":"gold",
+    		"zone":"us-east",
+    	}
+    }
+    "resources": {
+      "requests": {
+        "storage": "3Gi"
+      }
+    }
+  }
+}
+	
+```
+
+## Labelling PersistentVolumes
+
+A `PersistentVolumeSelector` on claim has the additional benefit of being able to bind to volumes that are manually created as opposed to dynamically through a provisioner.
+
+The same claim above (gold+east) can match on a PersistentVolume that is labelled similarly.
+
+```
+apiVersion: v1
```

Comment:
I understand.  It seems easy to accommodate.

What about "zone" and relying on the end user for the correct value "us-east-c1"?  

Override is disallowed by default (empty whitelist). It's up to the admin to accept responsibility? Do we need any validation now on inputs? 

How much of this is "must have" versus "good next enhancement"?  I don't think we're block on adding this.

I am open to it if it doesn't increase scope too much.